libraries/AP_Scripting/applets/idle_control_heli.lua

```diff
@@ -0,0 +1,241 @@
+-- idle_control.lua: a closed loop control throttle control while on ground idle (trad-heli)
+
+-- luacheck: only 0
+
+local PARAM_TABLE_KEY = 73
+local PARAM_TABLE_PREFIX = 'IDLE_'
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 30), 'could not add param table')
+
+
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+
+
+function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+    return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- parameters for idle control
+
+IDLE_GAIN_I = bind_add_param('GAIN_I', 1, 0.05)
+IDLE_GAIN_P = bind_add_param('GAIN_P', 2, 0.25)
+IDLE_GAIN_MAX = bind_add_param('GAIN_MAX', 3, 1)
+IDLE_MAX = bind_add_param('MAX', 4, 17)
+IDLE_RANGE = bind_add_param('RANGE', 5, 300)
+IDLE_SETPOINT = bind_add_param('SETPOINT', 6, 600)
+IDLE_RPM_ENABLE = bind_add_param('RPM_ENABLE', 7, 0)
+
+local H_RSC_IDLE = Parameter()
+H_RSC_IDLE:init('H_RSC_IDLE')
+
+local SERVO8_MAX = Parameter()
+SERVO8_MAX:init('SERVO8_MAX')
+
+local SERVO8_MIN = Parameter()
+SERVO8_MIN:init('SERVO8_MIN')
```

Comment:
You fetch the rsc channel later in the script but have the endpoints hard coded to servo 8 here. This should work:

```suggestion
local rsc_output = SRV_Channels:find_channel(31)
local SERVO_MAX = Parameter('SERVO' .. (rsc_output+1) .. '_MAX')
local SERVO_MIN = Parameter('SERVO' .. (rsc_output+1) .. '_MIN')
```

I have not tested.

Reply:
yes, I was looking indeed for a better way to retrieve endpoints through the servo function instead of hardcoding stuff (so to consider configs different from default).